workspaces/diff-engine/src/learn_shape/result.rs

```diff
@@ -762,8 +753,9 @@ mod test {
 
     let mut test_id_generator = TestIdGenerator::default();
 
-    let nullable_primitive_results =
-      collect_commands(nullable_primitive_observations.into_commands(&mut test_id_generator));
+    let nullable_primitive_results = collect_commands(
+      nullable_primitive_observations.into_commands(&mut test_id_generator, &JsonTrail::empty()),
+    );
     assert!(nullable_primitive_results.0.is_some());
     assert!(nullable_primitive_results.0.is_some());
```

Comment:
seems like there are a bunch of duplicates like this

Reply:
Echo! Leftover from #681, and we've gotta be good scouts :)

Reply:
fixed